vm/src/exceptions.rs

```diff
@@ -519,6 +519,15 @@ impl PyBaseException {
         let cls = zelf.class();
         format!("{}({})", cls.name(), repr_args.iter().format(", "))
     }
+
+    #[pymethod(magic)]
+    fn reduce(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyTupleRef {
```

Comment:
If it doesn't require type checking, this may be:
```suggestion
    fn reduce(zelf: PyObjectRef, vm: &VirtualMachine) -> PyTupleRef {
```

Reply:
```text
no method named `args` found for struct `core::PyObjectRef` in the current scope
method not found in `core::PyObjectRef`
```